### DISCUSSION TITLE
duplicate tables

### DISCUSSION THREAD

**Discussion by SaeedAlynya** [sqlitebrowser/sqlitebrowser#3115] [General]
i have a lot of tabels to create and all are the same duplicate can help a lot

**Comment by chrisjlocke** [comment]
![image](https://user-images.githubusercontent.com/7872218/178076904-ba84b701-441f-40f9-9732-92ed387734e1.png)

If you right-click on a table in the 'database structure' tab, you can select the option, 'copy create statement'.
This copies the SQL command to create this table into the clipboard.

![image](https://user-images.githubusercontent.com/7872218/178076986-2aa95a23-4d92-4111-b385-43ea8a969386.png)

If you then go to the 'Execute SQL' tab, and right-click and select 'Paste', it'll paste this command into the window.

![image](https://user-images.githubusercontent.com/7872218/178077057-7b10fd00-3ef0-4f10-9ec0-46a6ed33b05b.png)

You can then edit this as required, and then press [F5]

![image](https://user-images.githubusercontent.com/7872218/178077083-81bd6083-5377-4344-b604-20d2ff14f56e.png)

You can thank me now, or thank me later.  😁

**Comment by chrisjlocke** [comment]
For the avoidance of doubt, the 'create table' command is quite simple

    create table table-name (
    field-name fieldType
    )

In the above screenshot, DB4S has put 'orange' in double-quotes.  Eurgh.  You don't need to do this, unless you're using spaces in your names, which you shouldn't be doing anyway.  Same with the field names.

For field types, there are only three main ones.  text, integer and float.
So to create a table called 'fred' with three fields, aaa, bbb and ccc, just enter:

    create table fred (
    aaa integer,
    bbb text,
    ccc float)

You don't jhave to put each line on its own line, but it aids readability.

**Comment by chrisjlocke** [comment]
Ideally, each table should have a primary field, to ensure each row can be uniquely identified.  The easiest way is to use a 'primary key' identifier, so that SQLite uses a sequential number in that field ... 1 for record 1, 2 for record 2, etc.

    create table fred (
    aaa integer primary key,
    bbb text,
    ccc float)

**Comment by chrisjlocke** [comment]
> and all are the same 

Just as a 'er, are you sure?', why are you creating duplicate tables?  If (for example) you're storing names, you don't have to create a table to hold the names of people beginning with the letter A, another table for names of people beginning with the letter B, etc.  The whole point of a database is that the table holds a collection of similar pieces of data.  It's up to the fields to 'file' and determine that data.
A database table can hold millions of rows of data.  Its more efficient than a database holding a million tables.
